src/bevy_compat/keyboard.rs

```diff
@@ -0,0 +1,647 @@
+use std::{
+    collections::HashSet,
+    hash::{Hash, Hasher},
+};
+
+use bevy::{
+    input::{keyboard::KeyboardInput, ButtonState},
+    prelude::*,
+    window::PrimaryWindow,
+};
+use crossterm::event::KeyModifiers;
+
+use crate::{
+    event::{InputSet, KeyEvent},
+    kitty::KittyEnabled,
+};
+
+/// Pass crossterm key events through to bevy's input system.
+///
+/// You can use bevy's regular `ButtonInput<KeyCode>` or bevy_ratatui's
+/// `EventReader<KeyEvent>`.
+pub struct KeyboardPlugin;
+
+impl Plugin for KeyboardPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_plugins(bevy::input::InputPlugin)
```

Comment:
Using `bevy_ratatui` with a project using bevy's `DefaultPlugins` is a valid use case, and adding the InputPlugin separately here causes a panic in that case. Can we assume that users looking to ingest bevy input events will add the plugin themselves? Or is there an order of operations problem here?

Reply:
We need it in order to submit events. I'll make it conditional: only add it if it hasn't been added yet. This will mean that one should add `DefaultPlugins` before `RatatuiPlugins`.

Reply:
This means that even a user who doesn't need this bevy compatibility functionality will now be forced to include bevy_input, a dummy window will be created, etcetera. Can we conditionally add `KeyboardPlugin` based on whether InputPlugin is present? That way if bevy_input is not included, none of this runs, but if the user is interested in listening for normal bevy input, they have to add the InputPlugin as usual.

Reply:
How about we just take it out of the plugin group? Make it explicitly opt-in.

Reply:
In the `RatatuiPlugins` we have two flags already, `enable_kitty_protocol` and `enable_mouse_capture`. Could we add `enable_input_forwarding`? Makes it easy to flip the flag's default in the future if we want to make event forwarding the default experience.

Reply:
(Not going to hold up the PR on that though, just removing it from the plugin group is totally fine and convenience flags can be added later).

Reply:
I like this `enable_input_forwarding` idea.